### PR TITLE
ci: automate version bumps in module README.md files

### LINKED ISSUE #138: automate version bump in Module docs on new tag pushes
Add a GitHub action that creates a PR with a new tag for all README.md files and auto-approve it.

### PR DESCRIPTION
**Pull request by matifali**
closes #138

### CODE REVIEW

File: .github/workflows/update-readme.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Get the latest tag
+        id: get-latest-tag
+        run: echo "TAG=$(git describe --tags --abbrev=0 | sed 's/^v//')" >> $GITHUB_OUTPUT
+
+      - name: Run update script
+        run: ./update-readme.sh
```
[the remark this answers is not in the document]
Comment (matifali) [reply .github/workflows/update-readme.yaml:26]: yes, that's ideal. But how do we tag each module individually?

[the remark this answers is not in the document]
Comment (matifali) [reply .github/workflows/update-readme.yaml]: I updated to only update the modules which had changes between tags.

**Review by mafredri — APPROVED** [review]
One improvement suggestion, but otherwise looks good.

One thing to observe here is that a module can go e.g. from v1.0.0 to v1.5.2 in the blink of an eye since we're always bumping them by the coder/modules tag.

We shouldn't fix it in this PR, but eventually it could be nice for them to have their own versions.

**Comment by matifali** [comment]
@mafredri, is it possible for each module to have its own versions, given our current setup?

**Comment by matifali** [comment]
@mafredri, but our registry API needs to be changed then, currently, it is based on tags.

**Comment by mafredri** [comment]
> @mafredri, but our registry API needs to be changed then, currently, it is based on tags.

Don't think it would be a huge effort to do that, if we get some others on board of this being a good idea. Alternatively we can continue using tags but prefix them, like `vscode-web-v1.4.0`. I bet that would also require some small change to the registry.

**Comment by matifali** [comment]
Let's see what @bpmct thinks about this?

**Pull request merged**